src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.TruckProducer;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer<? extends Machine> mp;
+        if (type == Bulldozer.class) {
+            mp = new BulldozerProducer();
+        } else if (type == Excavator.class) {
+            mp = new ExcavatorProducer();
+        } else {
+            mp = new TruckProducer();
+        }
+        return (List<Machine>) mp.get();
```

Comment:
```suggestion
        if (type == Bulldozer.class) {
            return new ArrayList<>(new BulldozerProducer().get());
        } if (type == Excavator.class) {
            return new ArrayList<>(new ExcavatorProducer().get());
        } if (type == Truck.class) {
            return new ArrayList<>(new TruckProducer().get());
        }
        return Collections.emptyList();
```

Reply:
Thanks, that is definitely better solution.